src/buffer.c

```diff
@@ -41,6 +41,9 @@ int git_buf_try_grow(
 	if (buf->ptr == git_buf__oom)
 		return -1;
 
+	if (buf->asize == 0 && buf->size != 0)
+		return GIT_EINVALIDSPEC;
```

Comment:
I realize this is a ridiculous nitpick, but it seems weird that this error is called `INVALIDSPEC`.  It seems like it was intended for something like a refspec.

I wonder if changing this to `INVALIDINPUT` or just `INVALID` might make more sense for an error like this?